lib/cf-socket.c

```diff
@@ -146,7 +146,12 @@ static void nosigpipe(struct Curl_easy *data,
 #define nosigpipe(x,y) Curl_nop_stmt
 #endif
 
-#if defined(USE_WINSOCK) || \
+#if defined(USE_WINSOCK)
+#if defined(TCP_KEEPIDLE)&& defined(TCP_KEEPINTVL) && defined(TCP_KEEPCNT)
+#define WINSOCK_KEEP_SSO
```

Comment:
```suggestion
#define CURL_WINSOCK_KEEP_SSO
```
Suggestion to use something namespaced to curl to avoid collisions.